tests/scenarios/k8s/base/test-scripts/rdma_test.sh

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env bash
+
+# RDMA Over InfiniBand Test Script
+# Usage:
+#   - Server: ./rdma_test.sh server
+#   - Client: ./rdma_test.sh client leader
+
+set -euo pipefail
+
+# Check if the DEBUG env var is set to true
+if [ "${DEBUG:-false}" = "true" ]; then
+    set -x
+else
+    echo "Debug mode is disabled. Set env var 'DEBUG=true' to enable debug mode."
+fi
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
+# Set the RDMA port
+export PORT=18515
+
+function get_ib_devices() {
+    # Get the list of available IB devices. There is a possibility that the node
+    # has a NIC that shows up as mlx but is not an IB device. It could be using
+    # Ethernet at the link layer. So ignore such devices.
+    #
+    # One way to ignore such devices at NCCL level is to set "export
+    # NCCL_IB_HCA=^mlx5_8". So here we are ignoring the mlx5_8 device.
+    # See the failures like this: https://gist.github.com/surajssd/d65a9eb3844bfc49ccda3e84a0ff5a4b#file-nccl-failure-on-roce-mlx5_8-device-sh
+    IB_DEVICES=$(for dev in /sys/class/infiniband/*; do
+        hca=$(basename "$dev")
+        if [[ $(cat $dev/ports/1/link_layer 2>/dev/null) == "InfiniBand" ]]; then
+            echo "$hca"
+        fi
+    done)
```

Comment:
Have you looked into `ibv_devinfo -l`? This command automatically filters all NICs that don't have link type set to IB.